.circleci/test-scripts/integration-test/test-check-whether-to-run-job.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+SOURCE_DIR=$1
+SCRIPT_DIR="${SOURCE_DIR}/.circleci/test-scripts/integration-test"
+
+input=${2:-"$SCRIPT_DIR/TestInput.txt"}
+script="${SOURCE_DIR}/.circleci/integration-test/10-check-whether-to-run-job.sh"
+
+num_failures=0
+
+while read -r line; do
+    dockerized="$(echo "$line" | awk '{print $1}')"
+    image_family="$(echo "$line" | awk '{print $2}')"
+    trigger_source="$(echo "$line" | awk '{print $3}')"
+    BRANCH="$(echo "$line" | awk '{print $4}')"
+    TAG="$(echo "$line" | awk '{print $5}')"
+    expected_should_run="$(echo "$line" | awk '{print $6}')"
```

Comment:
Sorry I missed this in my previous reviews, but maybe we can use `read -a` to read values directly into an array and make it easier to assign these variables?
```suggestion
while read -r -a line; do
    dockerized="${line[1]}"
    image_family="${line[2]}"
    trigger_source="${line[3]}"
    BRANCH="${line[4]}"
    TAG="${line[5]}"
    expected_should_run="${line[6]}"
```

Reply:
Done